Jenkinsfile.promote

```diff
@@ -572,41 +568,47 @@ boolean isBranchProtected(String repo, String protectedBranch, String ghPath = "
 
 def setAllowForcePushes(String repo, String protectedBranch, String enabled, String ghPath = "../gh") {
     assertGithubCLI(ghPath)
-    //Use separate admin token credentials
-    withCredentials([string(credentialsId: getGhCredsID(), variable: 'GITHUB_TOKEN')]) {
-        //get current branch protection and remove allow force push
-        sh "${ghPath} api 'repos/${getGitAuthor()}/${repo}/branches/${protectedBranch}/protection' | " +
-           "jq 'del(.allow_force_pushes)' > protectionBefore.json"
-
-        //create new json based on current protection mapped as parameters
-        sh "jq \"${getProtectionMapScript()}\" protectionBefore.json | " +
-           "jq \". + {\"allow_force_pushes\":${enabled}}\" > protectionParameters.json"
-
-        //update protection on git
-        def allowForcePushEnabled = sh(script:
-                "${ghPath} api -XPUT 'repos/${getGitAuthor()}/${repo}/branches/${protectedBranch}/protection' " +
-                        "--input protectionParameters.json |" +
-                        " jq '.allow_force_pushes.enabled'", returnStdout: true).trim()
-        assert allowForcePushEnabled == enabled
-
-        //check that protection didn't changed except for allow_force_pushes
-        sh "${ghPath} api 'repos/${getGitAuthor()}/${repo}/branches/${protectedBranch}/protection' | " +
-                "jq 'del(.allow_force_pushes)' > protectionAfter.json";
-        def differences = sh (script: "diff protectionBefore.json protectionAfter.json | cat", returnStdout: true)
-        if (differences) {
-            error "Protection settings lost"+
-                    "\nBefore: "+
-                    "\n${readFile 'protectionBefore.json'} " +
-                    "\nAfter: "+
-                    "\n${readFile 'protectionAfter.json'} " +
-                    "\nDifferences: " +
-                    "\n${differences} " +
-                    "\nProtection parameters: " +
-                    "\n${readFile 'protectionParameters.json'} " +
-                    "Please rollback to Before state and update getProtectionMapScript"
+    try {
```

Comment:
catch error here would still be problematic or ? This method is called many times and still there is an operation done in-between which would fail if the `setAllowForcePushes` to true cannot be performed ...
I would put the whole content of `forcePushProtectedBranch` or `commitAndForcePushProtectedBranch` into the try-catch or even directly in the `Reset quickstarts branch` stage